docs/relationship_synthesis.md

```diff
@@ -0,0 +1,463 @@
+| OWNS                  | Indicates ownership of one entity to the other via their relationship.                                                         |
+| TEST                  | Used for testing purposes, might be a good option for experimentation.                                                         |
+
+#### Source & Target
+
+Source and Target define the methods for determining the GUIDs of both entities involved in the relationship. 
+To achieve this, we provide support for the following GUID Resolvers:
+
+1. `extractGuid`: Extracts the GUID from an existing attribute in the telemetry.
+2. `buildGuid`: Constructs the GUID using various pieces of information from the telemetry.
+3. `lookupGuid`: Performs a GUID lookup based on telemetry attributes. 
+
+##### extractGuid
+
+The `extractGuid` resolver is the simplest and allows specifying an attribute to use as the GUID.
+
+Example:
+
+```yaml
+  source:
+    extractGuid:
+      attribute: entity.guid
+  target:
+    extractGuid:
+      attribute: cluster.guid 
+```
+
+Another additional feature of the `extractGuid` is the ability to define the entity type.
+
+In the example below, you'll see how to define the entity type using an attribute in the target relationship.
+
+Example:
+
+```yaml
+  source:
+     extractGuid:
+        attribute: entity.guid
+  target:
+    extractGuid:
+      attribute: cluster.guid
+```
+
+Lastly, in special cases, you can provide a hardcoded value for the entityType. 
+We briefly discuss this in its dedicated section [here](#infra-na-for-extractguid).
+
+##### buildGuid
+
+The `buildGuid` resolver allows constructing the GUID from different pieces of information.
+
+Entity GUIDs are composed by 4 parts: `account`, `domain`, `type` and `identifier`. 
+
+You can read more about the usage of those fields on the GUID [here](entities/guid_spec.md).
+
+Example:
+
+```yaml
+source:
+  buildGuid:
+    account:
+      attribute: accountId
+      # or lookup: yes
+    domain:
+      value: EXT
+    type:
+      value: SERVICE
+    identifier:
+      fragments:
+      - value: "k8s:"
+      - attribute: "service.name"
+      - value: ":pod:"
+      hashAlgorithm: FARM_HASH
+```
+
+The `account` field can be defined using an attribute in the telemetry or by specifying a lookup key. 
+When a lookup key is defined, it will be searched within the accounts of the trusted account for the current data point (and used via candidate relationships).
+
+The `domain` and `type` fields can be either hardcoded values or attributes.
+
+The `identifier` can be constructed using one or multiple fragments. Each fragment can be a hardcoded string, an attribute, or a part of an attribute. 
+The identifier can be hashed using one of the supported hashing algorithms, with farmHash being the currently supported algorithm.
+
+If needed, a part of an attribute can be provided as a capture group in a regular expression.
```

Comment:
Couldn't come up w/ and example looking at the existing ones, do you know how to craft an example for the regular expression case?

Reply:
Decided to remove it for now, not available completely and still iterating on it on the backend.